tiantianXia/assignments/cyclicSort/cyclicSortEasy/cyclicSortEasy.js

```diff
@@ -35,3 +66,39 @@ console.log(`${cyclic_sort([1, 5, 6, 4, 3, 2])}`);
 
 // Space complexity #
 // The algorithm runs in constant space O(1).
+
+/**
+ * inputs: 
+ * @param {*} nums: an array that containing unique unsorted positive intergers.
+ * 
+ * outputs
+ * @returns nums: a sorted array
+ * 
+ * step 1: loop all the elements
+ * step 2: set j be the number of the nums[i] - 1 so that the function is able to compare the number at current position and number
+ *          in the j position which is the position that current element should at
+ * step 3: swap two element if two element is no equal
+ * 
+ * variable i: for iteration
+ * 
+ * while loop to traverse nums
+ *  set j as nums[i] - 1 so that we can compare that number at that position and swap
+ *    if current nums is not equal to the nums at j
+ *      swap current num and num at j
+ *    else increment i
+ * end loop
+ * return nums
+ * edge case nums[] => []
+ */
+//  const cyclic_sort = function (nums) {
+//   // TODO: Write your code here
+//   let i = 0;
+//   while(i < nums.length){
+//     const j = nums[i] - 1;
+//     if(nums[i] !== nums[j]){
+//       [nums[i], nums[j]] = [nums[j], nums[i]];
+//     }else
+//       i ++;
+//   }
+//   return nums;
+// };
```

Comment:
was this copy / paste or type out again? I am genuinely asking because I hope all students will take the time to type again!